containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceOverrideMapper.java

```diff
@@ -0,0 +1,327 @@
+        }
+
+        ResultSet resultset = session.execute("SELECT project_name FROM openshift_metrics.metrics_mappings WHERE project_id='%succeeded'");
+
+        Row row = (resultset == null) ? null: resultset.one();
+        String succeeded = (row == null) ? "false" : row.getString("project_name");
+
+        if (succeeded.equalsIgnoreCase("true")) {
+            log.debug("metrics_mapping table succeeded when it was created.");
+            return true;
+        } else {
+            log.debug("metrics_mapping table was created, but it did not succeed the last time.");
+            return false;
+        }
+    }
+
+    private void createMappingTable() {
+        session.execute("CREATE TABLE IF NOT EXISTS openshift_metrics.metrics_mappings (project_id text PRIMARY KEY, project_name text) WITH compaction = { 'class' : 'LeveledCompactionStrategy' };");
+    }
+
+    private Map<String,String> getMappingTable() {
+        // You only want to prepare the query once. Best to do it when you initialize the session.
+        PreparedStatement findMappings = session.prepare(
+                "SELECT project_id, project_name " +
+                        "FROM openshift_metrics.metrics_mappings " +
+                        "WHERE token(project_id) > ? AND token(project_id) <= ?");
+
+        Map<String,String> mappings = new HashMap<>();
+        if (hasMappingTable()) {
+            for (TokenRange tokenRange : getTokenRanges()) {
+                BoundStatement boundStatement = findMappings.bind().setToken(0, tokenRange.getStart())
+                        .setToken(1, tokenRange.getEnd());
+                ResultSet resultSet = session.execute(boundStatement);
+                resultSet.forEach(row -> mappings.put(row.getString(0), row.getString(1)));
+            }
+        }
+
+        mappings.remove("%succeeded");
+        return mappings;
+    }
+
+    private Set<TokenRange> getTokenRanges() {
+        Set<TokenRange> tokenRanges = new HashSet<>();
+        for (TokenRange tokenRange : session.getCluster().getMetadata().getTokenRanges()) {
+            tokenRanges.addAll(tokenRange.unwrap());
+        }
+        return tokenRanges;
+    }
+
+    private void writeMappingTable(Map<String,String> mappings) {
+        createMappingTable();
+
+        PreparedStatement pStatement = session.prepare("INSERT INTO openshift_metrics.metrics_mappings (project_id, project_name) VALUES(?,?);");
+
+        for (Map.Entry<String,String> entry: mappings.entrySet()) {
+            int max_retries = 3;
+            int tries = 0;
+            boolean succeeded = false;
+            while (!succeeded && tries < max_retries) {
+                try {
+                    log.infof("trying to write value to metrics_mapping table: %s %s", entry.getKey(), entry.getValue());
+                    session.execute(pStatement.bind(entry.getKey(), entry.getValue()));
+                    log.infof("wrote value to metrics_mapping table");
```

Comment:
Either remove this or change to debug. And if you change to debug, include the key/value written